python/cac_tripplanner/destinations/models.py

```diff
@@ -58,6 +62,47 @@ def __unicode__(self):
         return self.name
 
 
+class UserFlag(models.Model):
+    """Track flags set by mobile app users."""
+    class UserFlags(object):
+        been = 'been'
+        want_to_go = 'want_to_go'
+        not_interested = 'not_interested'
+        liked = 'liked'
+        none = ''
+
+        CHOICES = (
+            (been, 'Been'),
+            (want_to_go, 'Want to go'),
+            (not_interested, 'Not interested'),
+            (liked, 'Liked'),
+            (none, '')
+        )
+
+    # generic foreign key to abstract Attraction model
+    # see: https://docs.djangoproject.com/en/1.11/ref/contrib/contenttypes/#generic-relations
+    content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE)
+    object_id = models.PositiveIntegerField(null=False, db_index=True)
+    attraction = GenericForeignKey('content_type', 'object_id')
+    is_event = models.BooleanField(default=False, db_index=True)
+    timestamp = models.DateTimeField(default=now, editable=False, db_index=True)
+    user_uuid = models.UUIDField(editable=False, db_index=True)
+    flag = models.CharField(choices=UserFlags.CHOICES, max_length=32, db_index=True)
```

Comment:
The `flag` field needs to have `blank=True` set to allow for unsetting the flag. Without it, unsetting the flag causes an uncaught `ValidationError`.

Reply:
There is an explicit option for `none` in the `CHOICES set, which is an empty string, so still needing `blank` set here surprises me somewhat.